### PR TITLE
all: fix go vet warnings

### PR DESCRIPTION
**Pull request by fjl**
This PR fixes all current `go vet` warnings.
`go vet ./...` is now enabled on Travis CI.

### CODE REVIEW

**Comment by robotally** [comment]
| Vote | Count | Reviewers |
| :---: | :---: | :---: |
| :+1: | 0 |  |
| :-1: | 0 |  |

_Updated: Fri Apr 15 09:03:04 UTC 2016_

**Comment by fjl** [comment]
@karalabe PTAL

File: core/types/block.go [hunk 1]
```diff
@@ -330,6 +330,9 @@ func (b *Block) Extra() []byte            { return common.CopyBytes(b.header.Ext
 
 func (b *Block) Header() *Header { return CopyHeader(b.header) }
 
+// Body returns the non-header content of the block.
+func (b *Block) Body() *Body { return &Body{b.transactions, b.uncles} }
```
Comment (karalabe) [inline core/types/block.go]: Do we want the plain data or copies?

**Comment by karalabe** [comment]
LGTM :+1:

File: core/types/block.go (hunk 1 above)
Comment (fjl) [inline core/types/block.go]: `Block.Transactions` and `Block.Uncles` don't copy either.

Comment (karalabe) [inline core/types/block.go]: I know, that's why I thumbed up. The question was in general whether we'd like to do it?

Comment (fjl) [inline core/types/block.go]: Not sure, but it sounds like a waste of memory to me.

The change to "immutable" (copying) accessors was meant to be
made up for by reusing the memory obtained through the accessors.
But that turned out to be weird-looking so we don't do it.

**Comment by nagydani** [comment]
:+1:

**Pull request merged**